nssp/delphi_nssp/pull.py

```diff
@@ -39,26 +38,38 @@ def pull_nssp_data(socrata_token: str):
     ----------
     socrata_token: str
         My App Token for pulling the NWSS data (could be the same as the nchs data)
+    issue_date: Optional[str]
+        (patching mode only) YYYY-MM-DD formatted date of the issue to pull data for
+    source_dir: Optional[str]
+        (patching mode only) Path to the directory containing the source data csv files
+        The files in source_dir are expected to be named yyyy-mm-dd.csv
     test_file: Optional[str]
         When not null, name of file from which to read test data
+    logger:
+        Logger object
 
     Returns
     -------
     pd.DataFrame
         Dataframe as described above.
     """
-    # Pull data from Socrata API
-    client = Socrata("data.cdc.gov", socrata_token)
-    results = []
-    offset = 0
-    limit = 50000  # maximum limit allowed by SODA 2.0
-    while True:
-        page = client.get("rdmq-nq56", limit=limit, offset=offset)
-        if not page:
-            break  # exit the loop if no more results
-        results.extend(page)
-        offset += limit
-    df_ervisits = pd.DataFrame.from_records(results)
+    if not issue_date:
+        # Pull data from Socrata API
+        client = Socrata("data.cdc.gov", socrata_token)
+        results = []
+        offset = 0
+        limit = 50000  # maximum limit allowed by SODA 2.0
+        while True:
+            page = client.get("rdmq-nq56", limit=limit, offset=offset)
+            if not page:
+                break  # exit the loop if no more results
+            results.extend(page)
+            offset += limit
+        df_ervisits = pd.DataFrame.from_records(results)
+        logger.info(f"Grabbed {len(df_ervisits)} records from Socrata API")
+    else:
+        df_ervisits = pd.read_csv(f"{source_dir}/{issue_date}.csv")
```

Comment:
question: ah, we create a local copy of the data set using the "backup" script provided in the README, and then read a single issue-specific CSV from that archive? A few questions/comments

- Is it possible to include the archive-generation in the code here? We could even turn the "backup" script into a fn and call it as part of the patching module. I want to avoid having patching be a multi-step process. So I want to make sure the person patching doesn't need to remember to run the backup code first -- it should be automatic.
- I don't understand how the backup code is generating issue-specific datasets. Or are we doing post-processing somewhere?

Reply:
following up with that how big is the csv? Do we see any issues opening/processing into a dataframe?

Reply:
Each csv is just under 60mb, so no issue so far.